src/execution_impl.rs

```diff
@@ -13,88 +16,119 @@ use wasmer::{
+    address: &Address,
+    function: &str,
+    _args: T, // TODO: @adrien-zinger how do we pass function args in AssemblyScript?
+) -> Result<R> {
+    let value = call_module(env, address, function);
+    Ok(serde_json::from_str::<R>(&value.ret)?)
 }
 
 /// `Call` ABI called by the webassembly VM
+///
+/// Call an exported function in a WASM module at a given address
+///
 /// It take in argument the environment defined in env.rs
 /// this environment is automatically filled by the wasmer library
 /// And two pointers of string. (look at the readme in the wasm folder)
-fn call(env: &Env, address: i32, function: i32) -> i32 {
-    let memory = env.wasm_env.memory.get_ref().expect("initialized memory");
-    let addr_ptr = StringPtr::new(address as u32);
-    let func_ptr = StringPtr::new(function as u32);
-    let address = &addr_ptr.read(memory).unwrap();
-    let fnc = &func_ptr.read(memory).unwrap();
-    type GmSign = fn(&Address) -> Result<Bytecode>;
-    let get_module: GmSign = env.interface.get_module;
-    let module = &get_module(address).unwrap();
+pub fn call_module(env: &Env, address: &Address, function: &str) -> Response {
+    let get_module: fn(&Address) -> Result<Bytecode> = env.interface.get_module;
     sub_remaining_point(env, settings::METERING.call_price()).unwrap();
-    let value = exec(
+    let module = &get_module(address).unwrap();
+    exec(
         get_remaining_points(env),
         None,
         module,
-        fnc,
+        function,
         &[],
         &env.interface,
     )
-    .unwrap();
+    .unwrap() // TODO: Should return a Result<Response>
+}
+
+/// Raw call that have the right type signature to be able to be call a module
+/// directly form AssemblyScript:
+///
+#[doc = include_str!("../wasm/README.md")]
+pub fn assembly_script_call_module(env: &Env, address: i32, function: i32) -> i32 {
+    let memory = env.wasm_env.memory.get_ref().expect("initialized memory");
+    // TODO: replace all unwrap() by expect()
+    let addr_ptr = StringPtr::new(address as u32);
+    let func_ptr = StringPtr::new(function as u32);
+    let address = &addr_ptr.read(memory).unwrap();
+    let function = &func_ptr.read(memory).unwrap();
+    let value = call_module(env, address, function);
     let ret = StringPtr::alloc(&value.ret, &env.wasm_env).unwrap();
     ret.offset() as i32
 }
 
-/// Create an instance of VM from a module with a
-/// given intefrace, an operation number limit and a webassembly module
+/// `Print` ABI called by the webassembly VM
 ///
-fn create_instance(limit: u64, module: &[u8], interface: &Interface) -> Result<Instance> {
+/// An utility print function to write on stdout directly from AssemblyScript:
+pub fn assembly_script_print(env: &Env, arg: i32) {
+    let str_ptr = StringPtr::new(arg as u32);
+    // TODO: may be in wasmer-as create a pointer that keep memory?
+    println!(
+        "{}",
+        str_ptr
+            .read(env.wasm_env.memory.get_ref().expect("initialized memory"))
```

Comment:
typo : uninitialized ?

Reply:
Maybe the `expect()` could give a bit more context about what memory are we talking about (I imagine `massa-node` crashing with this obscure message) ...